chain-impl-mockchain/src/fragment/mod.rs

```diff
@@ -111,6 +114,7 @@ impl Fragment {
             Fragment::VoteCast(_) => FragmentTag::VoteCast,
             Fragment::VoteTally(_) => FragmentTag::VoteTally,
             Fragment::EncryptedVoteTally(_) => FragmentTag::EncryptedVoteTally,
+            Fragment::SmartContract(_) => FragmentTag::SmartContract,
```

Comment:
I think it would be more correct to call it `SmartContractDeploy` or something similar.

Reply:
And maybe have an enum to discriminate smart contract types, with EVM as the only currently defined variant.

Reply:
I changed the variant and the type as suggested, and updated the code.